src/contexts/NewDelivery.tsx

```diff
@@ -0,0 +1,189 @@
+import { createContext, ReactNode, useEffect, useState } from 'react'
+import { SubmitHandler } from 'react-hook-form'
+import useNewDeliverySteps from '@/hooks/useNewDeliverySteps'
+import { NewDeliveryContextValues } from '@/interfaces/newDeliveryContextValues'
+import { useTranslation } from 'react-i18next'
+
+interface NewDeliveryProviderProps {
+  children: ReactNode
+}
+
+export const NewDeliveryContext = createContext<NewDeliveryContextValues>({
+  currentStep: 1,
+  formsData: {},
+  openDialog: false,
+  onCloseDialog: () => {},
+  handleBack: () => {},
+  handleClickOpen: () => {},
+  handleSubmit: () => {},
+  updateStep4Item: () => {},
+  deleteStep4Item: () => {},
+  alertMessage: '',
+  isCompletedMove: false,
+  isExceedQuantity: false
+})
+
+interface MoveGood {
+  type: string // palettes | packages | pieces
+  quantity: number
+  zone: string
+}
+
+export default function NewDeliveryProvider({ children }: NewDeliveryProviderProps) {
+  const { t: translate } = useTranslation()
+  const [currentStep, setCurrentStep] = useState(1)
+  const [formsData, setFormsData] = useState<any>({})
+  const [openDialog, setOpenDialog] = useState(false)
+  const steps = useNewDeliverySteps()
+  const [step3Items, setStep3Items] = useState({ pallets: 0, packages: 0, pieces: 0 })
+  const [step4Items, setStep4Items] = useState<MoveGood[]>([{ type: '', quantity: 0, zone: '' }])
+  const [alertMessage, setAlertMessage] = useState('')
```

Comment:
maybe an array for the alert messages will work better. this way the user can see everything rather than the alert messages being rewritten every time until you see all is good.

Reply:
Sorry, but I don't understand an idea.

Reply:
After our conversation I understood and fixed it.